src/libraries/System.Runtime/tests/System/GCTests.cs

```diff
@@ -1058,11 +1058,59 @@ private static void AllocateArrayTooLarge()
             Assert.Throws<OutOfMemoryException>(() => GC.AllocateUninitializedArray<double>(int.MaxValue, pinned: true));
         }
 
-        [Fact]
-        private static void AllocateArrayRefType()
+        struct EmbeddedValueType<T>
```

Comment:
```suggestion
        [StructLayout(LayoutKind.Sequential)]
        struct EmbeddedValueType<T>
```
This should suppress the warning